zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -836,13 +837,35 @@ private void sendNote(NotebookSocket conn, HashSet<String> userAndRoles, Noteboo
       if (note.isPersonalizedMode()) {
         note = note.getUserNote(user);
       }
+      checkSequentialRunStatus(note);
       conn.send(serializeMessage(new Message(OP.NOTE).put("note", note)));
       sendAllAngularObjects(note, user, conn);
     } else {
       conn.send(serializeMessage(new Message(OP.NOTE).put("note", null)));
     }
   }
 
+  //check sequential running status (if NoteServer crash last time, it recover really status)
```

Comment:
`really` - do you mean `ready`?

Reply:
If the server crashes during RunAllParagraphs execution, then after switching on, the note will be in the same status. Here this case is just being processed.

Reply:
@felixcheung, I mean this function is to check the true running status of the paragraph.